src/NHibernate/Logging.cs

```diff
@@ -8,36 +8,90 @@
 		bool IsDebugEnabled { get; }
+		// Since v5.1
+		[Obsolete("Please use IsInfoEnabled() INHibernateLogger extension method instead.")]
 		bool IsInfoEnabled { get; }
+		// Since v5.1
+		[Obsolete("Please use IsWarnEnabled() INHibernateLogger extension method instead.")]
 		bool IsWarnEnabled { get; }
 
+		// Since v5.1
+		[Obsolete("Please use Error(string, params object[]) INHibernateLogger extension method instead.")]
 		void Error(object message);
+		// Since v5.1
+		[Obsolete("Please use Error(Exception, string, params object[]) INHibernateLogger extension method instead.")]
 		void Error(object message, Exception exception);
+		// Since v5.1
+		[Obsolete("Please use Error(string, params object[]) INHibernateLogger extension method instead.")]
 		void ErrorFormat(string format, params object[] args);
 
+		// Since v5.1
+		[Obsolete("Please use Fatal(string, params object[]) INHibernateLogger extension method instead.")]
 		void Fatal(object message);
+		// Since v5.1
+		[Obsolete("Please use Fatal(Exception, string, params object[]) INHibernateLogger extension method instead.")]
 		void Fatal(object message, Exception exception);
 
+		// Since v5.1
+		[Obsolete("Please use Debug(string, params object[]) INHibernateLogger extension method instead.")]
 		void Debug(object message);
+		// Since v5.1
+		[Obsolete("Please use Debug(Exception, string, params object[]) INHibernateLogger extension method instead.")]
 		void Debug(object message, Exception exception);
+		// Since v5.1
+		[Obsolete("Please use Debug(string, params object[]) INHibernateLogger extension method instead.")]
 		void DebugFormat(string format, params object[] args);
 
+		// Since v5.1
+		[Obsolete("Please use Info(string, params object[]) INHibernateLogger extension method instead.")]
 		void Info(object message);
+		// Since v5.1
+		[Obsolete("Please use Info(Exception, string, params object[]) INHibernateLogger extension method instead.")]
 		void Info(object message, Exception exception);
+		// Since v5.1
+		[Obsolete("Please use Info(string, params object[]) INHibernateLogger extension method instead.")]
 		void InfoFormat(string format, params object[] args);
 
+		// Since v5.1
+		[Obsolete("Please use Warn(string, params object[]) INHibernateLogger extension method instead.")]
 		void Warn(object message);
+		// Since v5.1
+		[Obsolete("Please use Warn(Exception, string, params object[]) INHibernateLogger extension method instead.")]
 		void Warn(object message, Exception exception);
+		// Since v5.1
+		[Obsolete("Please use Warn(string, params object[]) INHibernateLogger extension method instead.")]
 		void WarnFormat(string format, params object[] args);
 	}
 
-	public interface IInternalLogger2
+	// Required till look alike methods taking object are dropped.
+	[Obsolete("Implement and use NHibernate.INHibernateLogger")]
+	public interface ITransitionalInternaLogger : IInternalLogger
```

Comment:
Ok, I had to add some other dummy methods (but less dummy: they work) with this solution. Since we obsoleting passing an object as a message, and that the instance methods takes precedence other extensions even if extensions have a better matching signature, we need these methods for being able to call something not obsoleted in this case and without using code tricks.